### PR TITLE
Gdal version issue solved

### PR DESCRIPTION
**Pull request by ruchimotwaniBC**
[Description of PR]
Solves #869, the issue of gdal version.

Checklist:

* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: test/core/store/fs/test_registry.py [hunk 1]
```diff
@@ -341,7 +342,7 @@ def _assert_dataset_supported(
             requested_dtype_alias: Optional[str],
             expected_dtype_aliases: Set[str],
             expected_return_type: Union[Type[xr.Dataset],
-                                        Type[MultiLevelDataset]],
+            Type[MultiLevelDataset]],
```
Comment (AliceBalfanz) [inline test/core/store/fs/test_registry.py:345]: I believe the prior existing indent was correct.

File: xcube/core/timeseries.py [hunk 2]
```diff
@@ -59,7 +59,7 @@
     AGG_COUNT: CAN_COMPUTE
 }
 
-_CRS84 = pyproj.CRS.from_string('CRS84')
+_CRS84 = pyproj.CRS.from_string("OGC:CRS84")
```
Comment (AliceBalfanz) [inline xcube/core/timeseries.py:62]: We usually use single quotes for consitency, as far as possible. However, as you can tell from older code - we are mixing double quotes and single quotes... But would be great not to change existing single quotes into double quotes :)

File: test/core/store/fs/test_registry.py (hunk 1 above)
Comment (ruchimotwaniBC) [reply test/core/store/fs/test_registry.py:345]: Yes, true and I didnt notice this change. I do it :)

File: test/core/mldataset/test_base.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 from xcube.core.tilingscheme import TilingScheme
 from .helpers import get_test_dataset
 
+_CRS84 = 'OGC:CRS84'
```
Comment (AliceBalfanz) [inline test/core/mldataset/test_base.py:8]: @pont-us do we have a convention for naming constants? I feel like it is the first time I see a "_" prior to a constant name, but I might be wrong.

File: CHANGES.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 * Added a basic implementation of the draft version of OGC API - Coverages
 
 ### Fixes
+* Resolved the issue for CRS84 error due to latest version of gdal
```
Comment (AliceBalfanz) [inline CHANGES.md:8]:
```suggestion
* Resolved the issue for CRS84 error due to latest version of gdal (#869)
```

File: test/core/mldataset/test_base.py (hunk 3 above)
Comment (TonioF) [reply test/core/mldataset/test_base.py:8]: The leading underscore denotes constants that are only used within the module in which they are declared. This is used in some places in xcube and the plugins. I did not see any rule in our conventions that en- or discourages its use. Personally, I like it as it gives some order.

**Review by TonioF — COMMENTED** [review]
More of a comment than a full review: We have a module constants.py. It would be best to declare the CRS85 string there once and then have every place that requires it import it. If there ever would be another change, we would then just have to change the one place.

File: test/core/mldataset/test_base.py (hunk 3 above)
Comment (pont-us) [reply test/core/mldataset/test_base.py:8]: The leading underscore is recommended in PEP-8, so implicitly part of our conventions :)

**Comment by pont-us** [comment]
> More of a comment than a full review: We have a module constants.py. It would be best to declare the CRS85 string there once and then have every place that requires it import it. If there ever would be another change, we would then just have to change the one place.

In principle I agree that this would have been preferable, but I don't think that it's worth the additional effort to optimize it now. This PR already introduces module-level constants, which is at least a start. In the unlikely event that the identifier changes again, we can optimize to a global constant at that point.

File: xcube/core/tilingscheme.py [hunk 5]
```diff
@@ -36,11 +36,11 @@
     'EPSG:3857',
     'urn:ogc:def:crs:OGC:1.3:EPSG::3857',
 )
-GEOGRAPHIC_CRS_NAME = 'CRS84'
+GEOGRAPHIC_CRS_NAME = 'OGC:CRS84'
 GEOGRAPHIC_CRS_ALIASES = (
     'EPSG:4326',
     'urn:ogc:def:crs:OGC:1.3:EPSG::4326',
-    'CRS84',
+    'OGC:CRS84',
```
Comment (pont-us) [inline xcube/core/tilingscheme.py:43]:
```suggestion
    'OGC:CRS84',
    'CRS84',
```

This is a list of aliases recognized by the class, so we should keep the old identifier as well as adding the new one.

**Review by pont-us — CHANGES_REQUESTED** [review]
One change requested in code comments.

You should also update the xarray line in `environment.yml` to

```yaml
  - xarray >=2022.6,<2023.09.0
```

-- See #897 . This is of course unrelated to your changes but will make ensure that the tests run green.

I still think that the GitHub unit test failures are random and transient, so let's see if they go away on the next re-run. They do need to be fixed properly at some point, but not in this PR...

File: environment.yml [hunk 6]
```diff
@@ -44,7 +44,7 @@ dependencies:
   - tornado >=6.0
   - urllib3 >=1.26
   - werkzeug <2.2  # >=2.2 slows down S3 tests (deps: moto->flask->werkzeug)
-  - xarray >=2022.6
+  - xarray >=2022.6,<2023.09.0
```
Comment (AliceBalfanz) [inline environment.yml:47]: why is restricting xarray needed?

File: test/core/test_normalize.py [hunk 7]
```diff
@@ -21,6 +21,7 @@
 from xcube.core.normalize import normalize_coord_vars
 from xcube.core.normalize import normalize_dataset
 from xcube.core.normalize import normalize_missing_time
+from xcube.constants import CRS_CRS84
```
Comment (AliceBalfanz) [inline test/core/test_normalize.py:24]: typo? shouldn't it be CRS84?

File: xcube/constants.py [hunk 8]
```diff
@@ -64,6 +65,9 @@
     AUTHORITY["EPSG","4326"]]
 """
 
+CRS84 = 'OGC:CRS84'
+CRS_CRS84 = pyproj.crs.CRS.from_string('OGC:CRS84')
```
Comment (pont-us) [inline xcube/constants.py:69]:
```suggestion
CRS_CRS84 = pyproj.crs.CRS.from_string(CRS84)
```

I'm being pedantic, but this is a slight improvement :)

**Review by pont-us — APPROVED** [review]
Looks good! I have one very minor suggestion for improvement, but I don't need to review again, so approving.

File: xcube/constants.py (hunk 8 above)
Comment (ruchimotwaniBC) [reply xcube/constants.py:69]: Ya but make sense, can do it :)

File: environment.yml (hunk 6 above)
Comment (AliceBalfanz) [inline environment.yml:47]:
```suggestion
  - xarray >=2022.6,<2023.09.0 # because of issue https://github.com/dcs4cop/xcube/issues/897
```

**Review by AliceBalfanz — APPROVED** [review]
Approving, but please add a comment to the xarray restriction which explains why the versions are restricted :)

**Review by TejasMorbagal — APPROVED** [review]

**Pull request merged**